### ISSUE TITLE
[enhancement] AWS ECR support for containers required by addons

### ISSUE THREAD

**Issue by bogdando** [particuleio/terraform-kubernetes-addons#1326]
**Is your feature request related to a problem? Please describe.**
When deploying private EKS clusters on AWS, or just optimizing external traffic flows into EKS, it would be nice to have an option to reconfigure helm release resources for Helm charts, and kube manifest resources for k8s API, to use container images prepared for ECR.
And even more nice to have a programmatically defined list of such containers for each addon, and automate preparing ECR repositories, then copying  external images with skopeo.

**Describe the solution you'd like**
I have a [WIP branch](https://github.com/bogdando/teks/blob/main/QUICKSTART.md#demo-dev-cluster-un-this-fork) of tEKS and [another one](https://github.com/bogdando/terraform-kubernetes-addons/tree/dev) for this addons repo. Once my testing is done, I'll propose a PR.

**Describe alternatives you've considered**
Leave users on their own when preparing AWS ECR images for private EKS clusters, or to be brave and pull images from public sources to deploy on public EKS (though, Fargate node groups only support private clusters, for example...). Unfortunately the former wouldn't be possible with this terraform-kubernetes-addons automation for tEKS.

**Additional context**
It mostly done for my use case, I hope that would be a useful feature to contribute

**Comment by ArchiFleKs** [comment]
Hi @bogdando thanks for your contributions. The feature looks interesting (didn't know about the skopeo provider). I'm currently on holiday but happy to review when I'm back.

**Comment by bogdando** [comment]
Sure, feel free to include that. I'll also do my best to contribute changes in my fork back, eventually. Thank you for maintaining such great deployment automation tooling as tEKS!
